libcloud/compute/drivers/scaleway.py

```diff
@@ -0,0 +1,665 @@
+                                           region=region, paged=True)
+        availability = response.object['servers']
+
+        return sorted([self._to_size(name, sizes[name], availability[name])
+                       for name in sizes], key=lambda x: x.name)
+
+    def _to_size(self, name, size, availability):
+        min_disk = (_to_lib_size(size['volumes_constraint']['min_size'] or 0)
+                    if size['volumes_constraint'] else 25)
+        max_disk = (_to_lib_size(size['volumes_constraint']['max_size'] or 0)
+                    if size['volumes_constraint'] else min_disk)
+
+        extra = {
+            'cores': size['ncpus'],
+            'monthly': size['monthly_price'],
+            'arch': size['arch'],
+            'baremetal': size['baremetal'],
+            'availability': availability['availability'],
+            'max_disk': max_disk,
+            'internal_bandwidth': int(
+                (size['network']['sum_internal_bandwidth'] or 0) /
+                (1024 * 1024)),
+            'ipv6': size['network']['ipv6_support'],
+            'alt_names': size['alt_names'],
+        }
+
+        return NodeSize(id=name,
+                        name=name,
+                        ram=int((size['ram'] or 0) / (1024 * 1024)),
+                        disk=min_disk,
+                        bandwidth=int(
+                            (size['network']['sum_internet_bandwidth'] or 0) /
+                            (1024 * 1024)),
+                        price=size['hourly_price'],
+                        driver=self,
+                        extra=extra)
+
+    def list_images(self, region=None):
+        """
+        List available VM images.
+
+        :param region: The region in which to list images
+        (if None, use default region specified in __init__)
+        :type region: :class:`.NodeLocation`
+
+        :return: list of image objects
+        :rtype: ``list`` of :class:`.NodeImage`
+        """
+        response = self.connection.request('/images', region=region,
+                                           paged=True)
+        images = response.object['images']
+        return [self._to_image(image) for image in images]
+
+    def create_image(self, node, name, region=None):
+        """
+        Create a VM image from an existing node's root volume.
+
+        :param node: The node from which to create the image
+        :type node: :class:`.Node`
+
+        :param name: The name to give the image
+        :type name: ``str``
+
+        :param region: The region in which to create the image
+        (if None, use default region specified in __init__)
+        :type region: :class:`.NodeLocation`
+
+        :return: the newly created image object
+        :rtype: :class:`.NodeImage`
+        """
+        data = {
+            'organization': self.key,
+            'name': name,
+            'arch': node.extra['arch'],
+            'root_volume': node.extra['volumes']['0']['id']  # TODO check this
```

Comment:
Oh and we have another TODO here? What does it mean? Is someone really going to do this?

Reply:
This would be the second question I asked @sieben above - is the root volume always given as volume 0?